metadata-ingestion/src/datahub/ingestion/source/abs/config.py

```diff
@@ -113,9 +113,9 @@ def check_path_specs_and_infer_platform(
             raise ValueError("path_specs must not be empty")
 
         # Check that all path specs have the same platform.
-        guessed_platforms = set(
+        guessed_platforms = {
```

Comment:
imo explicitly saying `set` here is way more clear - I actually dislike this change

in general - I don't like any of the changes that replace `set()` calls with `{ x, y }` set literal syntax